### PR TITLE
Make binary recording memmap efficient II (using native memmaps)

### PR DESCRIPTION
**Pull request by h-mayorquin**
Larger change than #1741. Can be merged after or instead of it.

Uses native memmaps instead and adds an extra test.

Similar approach as the one used in https://github.com/SpikeInterface/spikeinterface/pull/1741.

### CODE REVIEW

**Comment by h-mayorquin** [comment]
This should close #1741

File: src/spikeinterface/core/binaryrecordingextractor.py [hunk 1]
```diff
@@ -156,25 +157,59 @@ def get_binary_description(self):
...
         start_frame: Union[int, None] = None,
         end_frame: Union[int, None] = None,
         channel_indices: Union[List, None] = None,
     ) -> np.ndarray:
-        traces = self._timeseries[start_frame:end_frame]
+        byte_offset = self.file_offset
+        dtype_size_bytes = self.dtype.itemsize
+        mmap_offset, array_offset = divmod(byte_offset, mmap.ALLOCATIONGRANULARITY)
+        data_size_bytes = dtype_size_bytes * self.num_samples * self.num_chan
+        mmmap_length = data_size_bytes + array_offset
```
Comment (samuelgarcia) [inline src/spikeinterface/core/binaryrecordingextractor.py:190]: Just for the pleasure to be picky : this could be in the init, no ?

Comment (h-mayorquin) [reply src/spikeinterface/core/binaryrecordingextractor.py]: Yep, makes sense, I moved it in the last commit.

That said, I think we could be even more efficient and only reserve the buffer that corresponds to the start_frame to end_frame. But let's do that in another PR.

**Review by samuelgarcia — APPROVED** [review]

File: src/spikeinterface/core/binaryrecordingextractor.py (hunk 1 above)
Comment (samuelgarcia) [reply src/spikeinterface/core/binaryrecordingextractor.py]: perfect!

**Comment by samuelgarcia** [comment]
Impeccable!

**Pull request merged**